.github/workflows/test.yaml

```diff
@@ -5,6 +5,8 @@ on:
     branches: [main]
   pull_request:
     branches: [main]
+  merge_group:
+    branches: [main]
```

Comment:
I am not sure if this works as intended. I have just merged another PR (#46 ) that wasn't rebased and I think it was right away merged without CI actions being before merge. I'd further test it.